checkov/terraform/plan_runner.py

```diff
@@ -0,0 +1,96 @@
+import dpath.util
+import logging
+import os
+
+from checkov.common.output.record import Record
+from checkov.common.output.report import Report
+from checkov.common.runners.base_runner import BaseRunner
+from checkov.runner_filter import RunnerFilter
+from checkov.terraform.checks.resource.registry import resource_registry
+from checkov.terraform.context_parsers.registry import parser_registry
+# Allow the evaluation of empty variables
+from checkov.terraform.plan_parser import parse_tf_plan
+LOG_LEVEL = os.getenv('LOG_LEVEL', 'WARNING').upper()
+logging.basicConfig(level=LOG_LEVEL)
+
+
+class Runner(BaseRunner):
+    check_type = "terraform_plan"
+
+    def __init__(self):
+        self.tf_definitions = {}
+        self.definitions_context = {}
+        self.template_lines = {}
+
+    block_type_registries = {
+        'resource': resource_registry,
+    }
+
+    def run(self, root_folder=None, external_checks_dir=None, files=None, runner_filter=RunnerFilter(),
+            collect_skip_comments=True):
+        report = Report(self.check_type)
+        self.tf_definitions = {}
+        parsing_errors = {}
+        if external_checks_dir:
+            for directory in external_checks_dir:
+                resource_registry.load_external_checks(directory, runner_filter)
+        if files:
+            files = [os.path.abspath(file) for file in files]
+            for file in files:
+                if file.endswith(".json"):
+                    tf_definitions, template_lines = parse_tf_plan(file)
+                    self.tf_definitions = tf_definitions
+                    self.template_lines = template_lines
+                    self.check_tf_definition(report, runner_filter)
+
+        report.add_parsing_errors(parsing_errors.keys())
+
+        return report
+
+    def check_tf_definition(self, report, runner_filter,
+                            ):
+
+        for full_file_path, definition in self.tf_definitions.items():
+            scanned_file = f"/{os.path.relpath(full_file_path)}"
+            logging.debug(f"Scanning file: {scanned_file}")
+            for block_type in definition.keys():
+                if block_type in ['resource']:
```

Comment:
@schosterbarak Instead of hardcoding `resource`, can we use the keys of the `block_type_registries` object?